distsql: "span used after call to Finish()" panic on navy

`navy 3` died with the following stack trace. Note that nodes `2` and `6` died with an OOM error before this. All nodes were running 6e824ae.
```
I170408 09:20:19.466354 102 vendor/google.golang.org/grpc/clientconn.go:806  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 192.168.1.5:26257: getsockopt: connection refused"; Reconnecting to {cockroach-navy-0002:26257 <nil>}
E170408 09:20:19.504352 378845296 sql/distsqlrun/server.go:216  [n3] flow ae27c9ce-d260-48d8-8a31-3bba19d9cf52 not found
E170408 09:20:19.508684 378845297 sql/distsqlrun/server.go:216  [n3] flow 0a1c1f78-4600-45b2-b96f-60ae4fd33b58 not found
panic: span used after call to Finish():
&{tracer:<nil> event:0xb8b350 Mutex:{state:0 sema:0} raw:{Context:{TraceID:6432722887789223840 SpanID:953811091574247686 Sampled:true Baggage:map[]} ParentSpanID:1028572818250674189 Operation:flow Start:{sec:63627240009 nsec:709284800 loc:0x2db4700} Duration:10000380620 Tags:map[] Logs:[]} numDroppedLogs:0}

goroutine 378845470 [running]:
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).maybeAssertSanityLocked(0xc45f1b15e0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:31 +0x301
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).Lock(0xc45f1b15e0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:25 +0x3f
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).LogFields(0xc45f1b15e0, 0xc4367d7cc0, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/span.go:123 +0xd3
github.com/cockroachdb/cockroach/pkg/util/tracing.(*TeeSpan).LogFields(0xc42f7ea140, 0xc4367d7cc0, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tee_tracer.go:212 +0x73
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x7f1f6eed31a8, 0xc48cc51d40, 0x1ab0001, 0x1ce4623, 0x8, 0xc4205d6d58, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:132 +0x299
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f1f6eed31a8, 0xc48cc51d40, 0x7f1f00000003, 0x2, 0x1d404f4, 0x54, 0xc4205d6f40, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:145 +0x22e
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f1f6eed31a8, 0xc48cc51d40, 0x1, 0xc400000003, 0x1d404f4, 0x54, 0xc42078c740, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:67 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Errorf(0x7f1f6eed31a8, 0xc48cc51d40, 0x1d404f4, 0x54, 0xc42078c740, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:125 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowRegistry).RegisterFlow.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_registry.go:184 +0x2bc
created by time.goFunc
        /usr/local/go/src/time/sleep.go:170 +0x44
```

--- a/pkg/sql/distsqlrun/flow_registry.go
+++ b/pkg/sql/distsqlrun/flow_registry.go
@@ -25,6 +25,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	opentracing "github.com/opentracing/opentracing-go"
 	"github.com/pkg/errors"
 )
 
@@ -175,8 +176,12 @@ func (fr *flowRegistry) RegisterFlow(
 				}
 			}
 			if numTimedOut != 0 {
+				// The span in the context might be finished by the time this runs. In
+				// principle, we could ForkCtxSpan() beforehand, but we don't want to
+				// create the extra span every time.
+				timeoutCtx := opentracing.ContextWithSpan(ctx, nil)
 				log.Errorf(
-					ctx,
+					timeoutCtx,
 					"flow id:%s : %d inbound streams timed out after %s; propagated error throughout flow",
 					id,
 					numTimedOut,

